src/middlewared/middlewared/plugins/pool.py

```diff
@@ -249,6 +249,45 @@ class Config:
             )
         ]
 
+    @accepts(Str('name', required=True))
+    def is_upgraded(self, name):
+        proc = subprocess.Popen(
+            f'zpool get -H -o value version {name}'.split(' '),
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf8'
+        )
+        res, err = proc.communicate()
+        if proc.returncode != 0 and self.middleware.call_sync('pool.query', [['name', '=', name]]):
+            return True
+        res = res.rstrip('\n')
+        try:
+            int(res)
+        except ValueError:
+
+            if res == '-':
+                proc = subprocess.Popen([
+                    "zpool",
+                    "get",
+                    "-H", "-o", "property,value",
+                    "all",
+                    name,
+                ], stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf8')
+                data = proc.communicate()[0].strip('\n')
+                for line in [l for l in data.split('\n') if l.startswith('feature') and '\t' in l]:
+                    prop, value = line.split('\t', 1)
+                    if value not in ('active', 'enabled'):
+                        return False
+                return True
+            else:
+                return False
+        else:
+            return False
+
+    @accepts(Str('pool', required=True))
+    def upgrade(self, pool):
```

Comment:
```
@accepts(Int('id', required=True))
@item_method
def upgrade(self, oid):
```

It should use id not name, to be consistent with the rest. Same above in is_upgraded